### PR TITLE
Remove error handling from loadObjects

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Minor cleanup - remove error handling from loadObjects - the processors now handle the exceptions themselves

Before
----------------------------------------
A whole lotta code for a little handling

After
----------------------------------------
poof

Technical Details
----------------------------------------
I wasn't sure about the debug_log_message part so I copied it into the processors, at risk of duplicating

Comments
----------------------------------------
This code really was our first foray into test-driven refactoring & exceptions - right back in 4.5 I or earlier

### CODE REVIEW

**Comment by seamuslee001** [comment]
@eileenmcnaughton Jenkins doesn't like your style here

**Comment by eileenmcnaughton** [comment]
@seamuslee001 - fixed

File: CRM/Core/Payment/AuthorizeNetIPN.php [hunk 1]
```diff
@@ -93,6 +93,7 @@ public function main($component = 'contribute') {
     }
     catch (CRM_Core_Exception $e) {
       Civi::log()->debug($e->getMessage());
+      CRM_Core_Error::debug_log_message($e->getMessage());
       echo 'Invalid or missing data';
```
Comment (mattwire) [inline CRM/Core/Payment/AuthorizeNetIPN.php:97]: @eileenmcnaughton Why are we changing to the older method here?

File: CRM/Core/Payment/PayPalIPN.php [hunk 2]
```diff
@@ -367,6 +367,7 @@ public function main() {
     catch (CRM_Core_Exception $e) {
       Civi::log()->debug($e->getMessage());
       echo 'Invalid or missing data';
+      CRM_Core_Error::debug_log_message($e->getMessage());
```
Comment (mattwire) [inline CRM/Core/Payment/PayPalIPN.php:370]: Now we're logging twice?

File: CRM/Core/Payment/PayPalProIPN.php [hunk 3]
```diff
@@ -487,6 +487,7 @@ public function main() {
     }
     catch (CRM_Core_Exception $e) {
       Civi::log()->debug($e->getMessage());
+      CRM_Core_Error::debug_log_message($e->getMessage());
```
Comment (mattwire) [inline CRM/Core/Payment/PayPalProIPN.php:490]: Logging twice?

**Comment by mattwire** [comment]
These are doing the same thing:
```
Civi::log()->debug($e->getMessage());
CRM_Core_Error::debug_log_message($e->getMessage());
```

To be safe you could prepend Civi with a \ to make sure it's found properly:
```
\Civi::log()->debug($e->getMessage());
```
but we don't need both lines (they resolve to the same function).

**Comment by eileenmcnaughton** [comment]
@mattwire are you sure they are exactly the same?

**Comment by seamuslee001** [comment]
@eileenmcnaughton @mattwire so Civi::log()->debug baisically means to use the psr_log service https://github.com/civicrm/civicrm-core/blob/master/Civi.php#L88 which for core purposes is an extension of the \PSR\Log\AbstractLogger class as per https://github.com/civicrm/civicrm-core/blob/master/Civi/Core/Container.php#L149 https://github.com/civicrm/civicrm-core/blob/master/CRM/Core/Error/Log.php#L18. All the Abstract logger class does is call the log function and passing in the respective level as the first param https://github.com/php-fig/log/blob/master/Psr/Log/AbstractLogger.php and eventually calls CRM_Core_Error::debug_log_message https://github.com/civicrm/civicrm-core/blob/master/CRM/Core/Error/Log.php#L61 the only real difference is that in the non Civi::log calls $priority will be NULL where as with the Civi::log() it will be PEAR_LOG_DEBUG

**Comment by eileenmcnaughton** [comment]
@seamuslee001 which sounds like you agree the extra log call should go....

**Comment by eileenmcnaughton** [comment]
I've removed them

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
test issue reported on infra

**Comment by seamuslee001** [comment]
Jenkins re test this please

**Comment by eileenmcnaughton** [comment]
@seamuslee001 OK to merge now?

**Comment by seamuslee001** [comment]
i think this looks fine to me

**Pull request merged**